x/mongo/driver/operation.go

```diff
@@ -590,6 +590,12 @@ func (op Operation) roundTrip(ctx context.Context, conn Connection, wm []byte) (
 		return nil, Error{Message: err.Error(), Labels: labels, Wrapped: err}
 	}
 
+	return op.readWireMessage(ctx, conn, wm)
+}
+
+func (op Operation) readWireMessage(ctx context.Context, conn Connection, wm []byte) ([]byte, error) {
```

Comment:
This is mostly the same logic that was originally in `roundTrip` but split off into a separate function so `ExecuteExhaust` can call it as well. The only difference is that it calls `SetStreaming` if a streaming connection is given and the response has `moreToCome`.